xbmc/platform/win32/Environment.cpp

```diff
@@ -0,0 +1,147 @@
+/*
+ *      Copyright (C) 2018 Team XBMC
+ *      http://kodi.tv
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+/**
+ * \file platform\win32\Environment.cpp
+ * \brief Implements CEnvironment win32 specified class functions.
+ */
+
+#include "platform/Environment.h"
+#include "platform/win32/CharsetConverter.h"
+#include <stdlib.h>
+#include <Windows.h>
+
+// --------------------- Internal Function ---------------------
+
+typedef int (_cdecl * wputenvPtr) (const wchar_t *envstring);
+
+/**
+ * \fn int CEnvironment::win32_setenv(const std::wstring &name, const std::wstring &value = L"",
+ *     updateAction action = autoDetect)
+ * \brief Internal function used to manipulate environment variables on win32.
+ *
+ * This function make all dirty work with setting, deleting and modifying environment variables.
+ *
+ * \param name   The environment variable name.
+ * \param value  (optional) the new value of environment variable.
+ * \param action (optional) the action.
+ * \return Zero on success, 2 if at least one external runtime update failed, 4 if process
+ * 		   environment update failed, 8 if our runtime environment update failed or, in case of
+ * 		   several errors, sum of all errors values; non-zero in case of other errors.
+ */
+int CEnvironment::win_setenv(const std::string &name, const std::string &value /* = "" */, enum updateAction action /* = autoDetect */)
+{
+  std::wstring Wname = KODI::PLATFORM::WINDOWS::ToW(name);
+  if (Wname.empty() || name.find('=') != std::wstring::npos)
+    return -1;
+  if ( (action == addOnly || action == addOrUpdateOnly) && value.empty() )
+    return -1;
+  if (action == addOnly && !getenv(name).empty() )
+    return 0;
+
+  std::wstring Wvalue = KODI::PLATFORM::WINDOWS::ToW(value);
+  int retValue = 0;
+
+  std::wstring EnvString;
+  if (action == deleteVariable)
+    EnvString = Wname + L"=";
+  else
+    EnvString = Wname + L"=" + Wvalue;
+
+  static const wchar_t *modulesList[] =
```

Comment:
we can get rid of this. It's not an issue anymore after we unified our crt depends.

Reply:
just to be clarify. does not it mater even if we update CRT to v141?